jerry-core/ecma/operations/ecma-regexp-object.cpp

```diff
@@ -230,6 +230,43 @@ ecma_op_create_regexp_object (ecma_string_t *pattern_p, /**< input pattern */
 } /* ecma_op_create_regexp_object */
 
 /**
+ * RegExp Canonicalize abstract operation
+ *
+ * See also: ECMA-262 v5, 15.10.2.8
+ *
+ * @return ecma_char_t canonicalized character
+ */
+static ecma_char_t __attr_always_inline___
+re_canonicalize (ecma_char_t ch, /**< character */
+                 bool is_ignorecase) /**< IgnoreCase flag */
+{
+  ecma_char_t ret_value = ch;
+
+  if (is_ignorecase)
+  {
+    /* 2. */
+    MEM_DEFINE_LOCAL_ARRAY (u, LIT_MAXIMUM_OTHER_CASE_LENGTH, ecma_char_t);
```

Comment:
Can't we just use a simple stack array?
ecma_char_t u[LIT_MAXIMUM_OTHER_CASE_LENGTH];

I don't think we need a dynamic alocation here.